safe_url_check/lib/safe_url_check.dart

```diff
@@ -159,8 +159,10 @@ Future<bool> _safeUrlCheck(
   if (response.isRedirect &&
       response.headers[HttpHeaders.locationHeader]!.isNotEmpty &&
       maxRedirects > 0) {
+    final loc = Uri.parse(response.headers[HttpHeaders.locationHeader]![0]);
+    final nextUri = loc.isAbsolute ? loc : url.resolveUri(loc);
```

Comment:
I've prepared a server + redirect logic for a test, but the test logic checks for local / private IPs and it always fails for such tests. Any preference how to suppress the IP checks for tests?

Reply:
Honestly, isn't it tempting to be lazy and just use: https://httpbin.org/redirect-to?url=status%2F200

Ignoring such local / private IP tests is kind of a bad idea, since the whole point is to avoid hitting private networks :D

I suppose one could do a one-off using a constant like `bool.fromEnvironment()` but that doesn't make testing easier, since you'll have to do `-D ..` something.

Let's either not test it, or use httpbin to not make our problems bigger than they have to be.

Reply:
> Honestly, isn't it tempting to be lazy and just use: https://httpbin.org/redirect-to?url=status%2F200

Already added that: https://github.com/google/dart-neats/pull/229/files#diff-d7c4d80be7934367cf516c2c7d1c65dd444fc2a10786ea20790381d3c361dea1R30

> Ignoring such local / private IP tests is kind of a bad idea, since the whole point is to avoid hitting private networks :D

I think it is not unreasonable to turn off some parts of the library for local testing - it feels that more coverage is better in this case. I'll check if I can implement something that is not exposed to the end user in any other way.

Reply:
Updated the PR with refactor + override for local testing. I can remove the commit if needed, but I think it is better to have it this way.